src/modules/chat_deleted_messages/index.js

```diff
@@ -0,0 +1,96 @@
+const watcher = require('../../watcher');
+const twitch = require('../../utils/twitch');
+const settings = require('../../settings');
+
+const CHAT_CONTAINER = '.chat-room__container';
+const CHAT_LINE_SELECTOR = '.chat-line__message';
+const CHAT_LINE_DELETED_CLASS = 'chat-line__message--deleted';
+
+function chatOberserverHandler(mutations) {
+    const showDeleted = settings.get('showDeletedMessages');
+    const hideDeleted = settings.get('hideDeletedMessages');
+    if (!showDeleted && !hideDeleted) {
+        // return early
+        return;
+    }
+    const elementsToReattach = [];
+    const elementsToRemove = [];
+
+    mutations.forEach(mutation => {
+        if (mutation.type === 'childList' && mutation.target.matches(CHAT_LINE_SELECTOR)) {
+            mutation.addedNodes.forEach(node => {
+                if (!node.__is_prevented && node.classList && node.classList.contains(CHAT_LINE_DELETED_CLASS)) {
```

Comment:
i've tried something similar to this in the past.. it wasn't performant enough and caused the chat to flicker.. we might want to monkeypatch and handle deletes instead

Reply:
I've used the mutation observer on 30k+ channels daily with tampermonkey, and so far I've had no problems with chrome 61 and 62 on my potato computer. 

Was this flickering consistent?